cap/supervisor_options.go

```diff
@@ -52,14 +52,19 @@ func WithNotifier(en EventNotifier) Opt {
 	}
 }
 
+// withNodes is used internally by a few functions in this API.
+func withNodes(nodes []Node) BuildNodesFn {
+	emptyCleanupResources := func() error { return nil }
+	return func() ([]Node, CleanupResourcesFn, error) {
+		return nodes, emptyCleanupResources, nil
+	}
+}
+
 // WithNodes allows the registration of child nodes in a SupervisorSpec. Node
 // records passed to this function are going to be supervised by the Supervisor
 // created from a SupervisorSpec.
 //
 // Check the documentation of NewSupervisorSpec for more details and examples.
 func WithNodes(nodes ...Node) BuildNodesFn {
-	emptyCleanupResources := func() error { return nil }
-	return func() ([]Node, CleanupResourcesFn, error) {
-		return nodes, emptyCleanupResources, nil
-	}
+	return withNodes(nodes)
```

Comment:
I do not see the reason for this added indirection 🤷‍♂️

Reply:
Let me see if I can remove it, I was a bit tired when I did this and I may have a better way to go about it